vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/StaffSearchManager.java

```diff
@@ -0,0 +1,184 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.TextContentBlockRepository;
+import edu.asu.diging.vspace.core.model.impl.Module;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+/**
+ * This class has all the methods with the business logics for searching a word
+ * in staff page.
+ * 
+ * @author Avirup Biswas
+ *
+ */
+@Service
+public class StaffSearchManager implements IStaffSearchManager {
+
+    @Value("${page_size}")
+    private int pageSize;
+
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private ISlideManager slideManager;
+
+    @Autowired
+    private TextContentBlockRepository textContentBlockRepo;
+
+    /**
+     * Method to return the requested spaces whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      Spring will just return an empty dataset if a page that is
+     *                   greater than the total number of pages is returned. So, the
+     *                   page given (except when it's < 1, then set it to 1) and
+     *                   then checking if the total page count is smaller than the
+     *                   given one (if so, make another request). In most cases the
+     *                   page number will be within the range of pages, so we would
+     *                   need only one db call. Only in cases where the page number
+     *                   is wrong, would a second one be needed.The page number
+     *                   starts from 1. if page<1, 1st page is returned, if
+     *                   page>total pages,last page is returned.
```

Comment:
the implementation details of the methods should be as a ocmment in the code; here it should just tell whoever uses the method what to pass in (page number based 1)